session/payment/session_payments.go

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package payment
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	"github.com/mysteriumnetwork/node/session/balance"
+	"github.com/mysteriumnetwork/node/session/promise"
+	"github.com/mysteriumnetwork/payments/promises"
+)
+
+// PeerPromiseSender knows how to send a promise message to the peer
+type PeerPromiseSender interface {
+	Send(promise.Message) error
+}
+
+// PromiseTracker keeps track of promises
+type PromiseTracker interface {
+	AlignStateWithProvider(providerState promise.State) error
```

Comment:
I personally feel word 'Aligned' as a curse :), but if you don't afraid it, you can use it ;)